### PR TITLE
Add import checking with isort

### PR DESCRIPTION
**Pull request by paltman**
#### What's this PR do?

"Lint" the imports with `isort`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
       - checkout
       - restore_cache:
           keys:
-            - v2-lint-deps
+            - v3-lint-deps
```
Comment (blueyed) [inline .circleci/config.yml]: What about adding `{{ checksum "tox.ini" }}` instead?
Makes sense in general, too.

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,22 +32,27 @@ jobs:
...
       - run:
           name: lint code
           command: |
             . venv/bin/activate
             flake8 pinax
+      - run:
+          name: check imports
+          command: |
+            . venv/bin/activate
+            isort pinax/**/*.py -c --diff
```
Comment (blueyed) [inline .circleci/config.yml]: I suggest using `isort --recursive --check-only --diff pinax` instead.

File: tox.ini [hunk 3]
```diff
@@ -42,7 +42,9 @@ commands =
 [testenv:checkqa]
 basepython = python3.6
 commands =
-    flake8 --inline-quotes "double" pinax
+    flake8 pinax
+    isort pinax/**/*.py -c
```
Comment (blueyed) [inline tox.ini]: Duplicated.

I think it is better to just run `tox -e checkqa` as a single job.

File: tox.ini [hunk 4]
```diff
@@ -42,7 +42,9 @@ commands =
...
 deps =
     flake8 == 2.5.4
     flake8-quotes == 0.11.0
+    isort == 4.2.15
```
Comment (blueyed) [inline tox.ini]: Deps could be centralized as an `extras_require` for e.g. `qa` in setup.py.

**Comment by paltman** [comment]
woah, lots different results with those flags

File: tox.ini (hunk 4 above)
Comment (blueyed) [reply tox.ini]: See https://github.com/Vimjas/covimerage/blob/c8116278c3b00e80c9be6c3e30abe9ef6dab23c6/setup.py#L94-L106.

**Comment by paltman** [comment]
back to where we started.  isort locally says there is nothing wrong with the webhooks modules but on the CI server it complains.  Also i think the run on the CI server is wrong.  `st` comes after `si`.

**Comment by blueyed** [comment]
Make sure it is using the same config.

**Comment by paltman** [comment]
It is.  they are using `tox.ini`

**Comment by paltman** [comment]
Once again, I'm baffled by `isort`'s behavior.

Why does it want there to be a blank line between `import stripe` and `from mock` where it doesn't what separation between `import stripe` and `from six`.  It also wants `from mock` after `import stripe`.

**Comment by blueyed** [comment]
(recognized or not) first/second/third party sections?!

**Comment by paltman** [comment]
Perhaps, maybe I just don't understand them (yet) but I would have thought that `stripe`, `six`, and `mock` where all the same.  They are all pip installed.

**Comment by paltman** [comment]
I guess this is a lie:



:)

Also, tests in python3 envs are now failing with: 

```UnicodeEncodeError: 'ascii' codec can't encode character '\xe1' in position 22: ordinal not in range(128)```

Maybe it was a mistake to have tox run things on circleci.  Nothing but problems.

**Comment by blueyed** [comment]
You need to pass some envs to tox.

**Comment by blueyed** [comment]
Not sure about UnicodeEncodeError, but tox is in general a good layer of extra isolation IMHO.

**Comment by blueyed** [comment]
IIRC I've put the token for https://github.com/Vimjas/covimerage/ into the env.

**Comment by paltman** [comment]
So the `UnicodeEncodeError` was coming from setting various `LANG` vars with `setenv` in `tox.ini`.  Removing those got things to pass.  Not sure why it worked locally but I'm happy that less is more. :)

The token issue was just setting some `passenv` for `CIRCLE_*`.

**Pull request merged**